test/e2e/storage/pv_protection.go

```diff
@@ -129,3 +128,15 @@ var _ = utils.SIGDescribe("PV Protection", func() {
 		framework.WaitForPersistentVolumeDeleted(client, pv.Name, framework.Poll, framework.PVDeletingTimeout)
 	})
 })
+
+func check_pv_protection_finalizer(client clientset.Interface, pv *v1.PersistentVolume) {
+	err := wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+		pv, err := client.CoreV1().PersistentVolumes().Get(pv.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred(), "While getting PV status")
```

Comment:
ideally, I would restructure this function such that, it just returns the error. Actual expectations will still be part of test scenario above.

Reply:
Yeah, thanks again. I have done that.

Reply:
done

Reply:
Sorry, I misunderstood your comment.
TBH, I am not sure how to restructure to return this case error to the method caller side in this case.

Reply:
You can do something like:
```
pv, err := client.CoreV1()...
if err != nil {
    return false, err
}
```
Which would terminate the waiting function early with an error.

Reply:
Thanks for your help, will apply it :-)